src/ArraySerializable.php

```diff
@@ -68,6 +68,10 @@ public function hydrate(array $data, $object)
         }
 
         if (is_callable([$object, 'exchangeArray'])) {
+            if(is_callable([$object, 'getArrayCopy'])) {
+                $original = $object->getArrayCopy($object);
+                $replacement = array_merge($original, $replacement);
```

Comment:
I would use zend-stdlib's `ArrayUtils::merge()` here, to ensure any nested arrays are merged in a usable fashion.

Reply:
@weierophinney, I’m afraid this change makes it impossible to remove items from collections. Consider the following code:

```php
$original = [
    'headline' => 'A new post',
    'tags'     => ['zend-hydrator']
];

$replacement = [
    'headline' => 'An updated post',
    // I’d like to remove the “zend-hydrator” tag
    'tags'     => []
];

// array_merge correctly removes that tag, simply replacing $original['tags']
// with $replacement['tags']
var_dump(array_merge($original, $replacement));

// ArrayUtils::merge doesn’t allow to remove any item from the tags collection
var_dump(Zend\Stdlib\ArrayUtils::merge($original, $replacement));
```

Reply:
@gscscnd 
Please open a new issue report for this problem. Thanks!